App/Sources/Scenes/Settings/SettingsViewController.swift

```diff
@@ -2,44 +2,94 @@
 //  SettingsViewController.swift
 //  App
 //
-//  Created by Yu Tawata on 2019/05/05.
+//  Created by Yu Tawata on 2019/07/13.
 //
 
 import Foundation
 import UIKit
+import Pipeline
+import BitriseAPI
 import Shared
+import Domain
 
 class SettingsViewController: UIViewController, Instantiatable {
     struct Dependency {
+        let store: StoreProtocol
+        let presenter: SettingsViewPresenterProtocol
     }
 
+    enum SectionKind: Int {
+        case tokens
+        case app
+    }
+
+    enum ItemKind: Hashable {
+        case token(String)
+        case version(String)
+    }
+
+    @IBOutlet weak var contentView: UIView!
     @IBOutlet weak var tableView: UITableView! {
         didSet {
-            tableView.tableFooterView = UIView()
-            tableView.dataSource = self
+            apply(tableView, mainSceneStyle)
             tableView.delegate = self
         }
     }
 
+    private lazy var dataSource = UITableViewDiffableDataSource<SectionKind, ItemKind>(tableView: tableView) { (tableView, indexPath, itemKind) -> UITableViewCell? in
+        switch SectionKind(rawValue: indexPath.section)! {
+        case .tokens:
+            let cell = UITableViewCell()
+            cell.textLabel
+            return cell
+        case .app:
+            let cell = UITableViewCell()
+            cell.textLabel
+            return cell
+        }
+    }
+//    (tableView: tableView) { (tableView, indexPath, item) -> UICollectionViewCell? in
+//        switch SectionKind(rawValue: indexPath.section)! {
+//        }
+//        let cell = BitriseBuildStatusCell.dequeue(for: indexPath, from: collectionView)
+//        cell.configure(.init(child: build))
+//        return UITableViewCell()
+//    }
+    private var dependency: Dependency!
+
     override func viewDidLoad() {
         super.viewDidLoad()
+    }
 
-        navigationItem.title = "Settings"
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        dependency.presenter.dispatch(.load).execute(.init(store: dependency.store))
+
+        dependency.presenter.subscribe(configure(_:))
     }
 
-    func inject(dependency: SettingsViewController.Dependency) {
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        dependency.presenter.unsubscribe()
     }
-}
 
-extension SettingsViewController: UITableViewDataSource {
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 0
+    func inject(dependency: SettingsViewController.Dependency) {
+        self.dependency = dependency
     }
 
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        return UITableViewCell()
+    private func configure(_ state: SettingsScene.State) {
+        let snapshot = apply(NSDiffableDataSourceSnapshot<SectionKind, ItemKind>(), { (snapshot) in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/yuta24/Cimon/blob/1c3e07373f4474a7fd0070b823e8732b727f8267/App/Sources/Scenes/Settings/SettingsViewController.swift#L83"></span>Unused parameter “snapshot” in a closure should be replaced with _.<br />
<code>unused_closure_parameter</code> <code>SettingsViewController.swift:83</code></td>
    </tr>
  </tbody>
</table>